tracing-opentelemetry/src/span_ext.rs

```diff
@@ -86,6 +120,26 @@ impl OpenTelemetrySpanExt for tracing::Span {
         });
     }
 
+    fn add_link(&self, cx: Context) {
+        let mut cx = Some(cx);
+        self.with_collector(move |(id, collector)| {
+            if let Some(get_context) = collector.downcast_ref::<WithContext>() {
+                get_context.with_context(collector, id, move |builder, _tracer| {
+                    if let Some(cx) = cx.take() {
+                        let follows_context = cx.span().span_context().clone();
+                        let follows_link =
+                            opentelemetry::trace::Link::new(follows_context, Vec::new());
+                        if let Some(ref mut links) = builder.links {
+                            links.push(follows_link);
+                        } else {
+                            builder.links = Some(vec![follows_link]);
+                        }
+                    }
+                });
+            }
```

Comment:
i think that if we borrowed the `Context` parameter, as in my above suggestion, the implementation would be a bit simpler:
```suggestion
    fn add_link(&self, cx: &Context) {
        self.with_collector(|(id, collector)| {
            if let Some(get_context) = collector.downcast_ref::<WithContext>() {
                get_context.with_context(collector, id, |builder, _tracer| {
                    let follows_context = cx.span().span_context().clone();
                    let follows_link =
                        opentelemetry::trace::Link::new(follows_context, Vec::new());
                    if let Some(ref mut links) = builder.links {
                        links.push(follows_link);
                    } else {
                        builder.links = Some(vec![follows_link]);
                    }
                });
            }
```